app/src/main/java/com/example/itunesapp/adapter/SearchAdapter.kt

```diff
@@ -0,0 +1,30 @@
+package com.example.itunesapp.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.itunesapp.databinding.ListItemBinding
+import com.example.itunesapp.model.RemoteDataModel
+
+class SearchAdapter(private val songList: ArrayList<RemoteDataModel>) :RecyclerView.Adapter<SearchAdapter.ViewHolder>() {
+
+    inner class ViewHolder(val binding: ListItemBinding) : RecyclerView.ViewHolder(binding.root)
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val binding = ListItemBinding
+            .inflate(LayoutInflater.from(parent.context),parent,false)
+        return ViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        with(holder){
+            with(songList[position]){
+                binding.songName.text = collectionName
+            }
+        }
```

Comment:
We would generally use with if we need to access multiple values and functions on a particular object. Otherwise is just creates unnecessary nesting.